src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -83,13 +83,18 @@ public static class TagHelperDescriptorFactory
             bool designTime)
         {
             TagHelperUsageDescriptor typeUsageDescriptor = null;
+            string outputElementHint = null;
 
-#if !DNXCORE50
             if (designTime)
             {
+                var outputElementHintAttribute =
+                    typeInfo.GetCustomAttribute<OutputElementHintAttribute>(inherit: false);
+
+                outputElementHint = outputElementHintAttribute?.OutputElement;
```

Comment:
Could revamp `TagHelperUsageDescriptor` to be the design time stuff property. Would just have to add an extra type maybe like `TagHelperElementUsageDescriptor` which would contain `OutputElementHint` for `TagHelperDescriptor`s. `TagHelperAttributeDescriptor`s would continue to use `TagHelperUsageDescriptor`.

What do you guys think?